components/Checkout/ExtraFieldsForm.js

```diff
@@ -19,7 +20,9 @@ function ExtraFieldsForm() {
   const { setCurrentStep, nextStepFrom } = useCheckoutDispatch();
 
   if (extrafields.length === 0) {
-    setCurrentStep(nextStepFrom("extrafields"));
+    useEffect(() => {
+      setCurrentStep(nextStepFrom("extrafields"));
+    }, [extrafields]);
     return null;
   }
```

Comment:
You'll need to move the condition inside of the `useEffect` call.

Reply:
Sorry I meant the `if (extrafields.length === 0)` needs to be inside the `useEffect` statement, as you can have side effects that happen depending on whether a condition passes, one of the rules of hooks.

Reply:
GAH sorry my bad i read it as "remove" not "move"